spring-rabbit/src/main/java/org/springframework/amqp/rabbit/annotation/RabbitListenerAnnotationBeanPostProcessor.java

```diff
@@ -779,4 +761,56 @@ private MessageHandlerMethodFactory createDefaultMessageHandlerMethodFactory() {
 
 	}
 
+	private class TypeMetadataHolder {
```

Comment:
As I said: the target result is idempotent, so, I won't worry if my `TypeMetadata` is filled concurrently, even if we have some extra computation for nothing. But still it might be better then wait for another thread.